src/extremes_based/extremesdim.jl

```diff
@@ -206,3 +207,44 @@ function extremevaltheory_local_dim_persistence(
     Δ, θ = extremevaltheory_local_dim_persistence(logdist, p; kw...)
     return Δ, θ
 end
+
+
+
+"""
+BMextremedimensions(x:: StateSpaceSet) -> D1, θ
+    Computation of the local dimensions D1 and the extremal index θ for each observation in the 
+    trajectory x. This function uses the block maxima approach: divides the data in blocks of 
+    length ⌊√(N-1)⌋, where N is the number of data and ⌊⋅⌋ is the floor function, and takes the maximum
```

Comment:
Shouldn't the user be able to specify the length, that defaults to the value `⌊√(N-1)⌋`?

Also, please remove the `⌊⋅⌋` stuff. Just use `floor(√(N-1))`. Julia already has a floor function, the less new symbols we have to introduce, the better.

Reply:
That could be a possibility, in that case it would be more straightforward to let the user specify the block length, or the number of blocks, but that still leaves the problem that the total amount of data that the function uses is blockSize*numberOfBlocks + 1, so the rest can't be used. Also there must be some equilibrium between the block size and the number of blocks, I thought that forcing it to be ⌊√(N-1)⌋ for both was the most straightforward way of ensuring that most of the data is used, and that there are no tiny blocks, for example if N happens to be something like p*2+1 with p a prime number. Also, if the specified length is less than the amount of data available, you can only lose data, without apparent gain except runtime.

Reply:
Sure. Just make the block number an optional argument with this default value new type `BlockMaximaMM` . In the documentation string of the new type `BlockMaximaMM`  you can provide the information you provided here to help the users make an informed decision.